llvm/include/llvm/IR/DIExpressionOptimizer.h

```diff
@@ -0,0 +1,81 @@
+//===- llvm/IR/DIExpressionOptimizer.h - Constant folding of DIExpressions --*-
+//C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Declarations for functions to constant fold DIExpressions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_IR_DIEXPRESSIONOPTIMIZER_H
+#define LLVM_IR_DIEXPRESSIONOPTIMIZER_H
+
+#include "llvm/IR/DebugInfoMetadata.h"
+
+using namespace llvm;
+
+/// Returns true if the Op is a DW_OP_constu.
+bool isConstantVal(uint64_t Op);
```

Comment:
These functions probably don't need to be all public. If you just move the entry-level function into DIExpressionOptimizer.cpp then these can all be static methods?

Reply:
And then you probably don't even need a new header file.